ravenframework/SupervisedLearning/DMD/DMDBase.py

```diff
@@ -0,0 +1,373 @@
+        specifying input of cls.
+    """
+    specs = super().getInputSpecification()
+    specs.addSub(InputData.parameterInputFactory("light", contentType=InputTypes.BoolType,
+                                                 descr=r"""TWhether this instance should be light or not. A light instance uses
+                                                 less memory since it caches a smaller number of resources.""", default=False))
+    specs.addSub(InputData.parameterInputFactory("pivotParameter", contentType=InputTypes.StringType,
+                                                 descr=r"""defines the pivot variable (e.g., time) that represents the
+                                                 independent monotonic variable""", default="time"))
+    specs.addSub(InputData.parameterInputFactory("reductionMethod", contentType=InputTypes.makeEnumType("reductionMethod", "reductionMethodType",
+                                                                                                        ["svd", "correlation_matrix", "randomized_svd"]),
+                                                 descr=r"""the type of method used for the dimensionality reduction.Available are:
+                                                  \begin{itemize}
+                                                    \item \textit{svd}, single value decomposition
+                                                    \item \textit{svd}, randomized single value decomposition
+                                                    \item \textit{correlation\_matrix}, correlation-based reduction.
+                                                  \end{itemize}""", default="svd"))
+    specs.addSub(InputData.parameterInputFactory("reductionRank", contentType=InputTypes.IntegerType,
+                                                 descr=r"""defines the truncation rank to be used for the reduction method.
+                                                 Available options are:
+                                                 \begin{itemize}
+                                                 \item \textit{-1}, no truncation is performed
+                                                 \item \textit{0}, optimal rank is internally computed
+                                                 \item \textit{$>1$}, this rank is going to be used for the truncation
+                                                 \end{itemize}""", default=0))
+    return specs
+
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+    settings, notFound = paramInput.findNodesAndExtractValues(['pivotParameter','light', 'reductionRank', 'reductionMethod'])
+    # notFound must be empty
+    assert(not notFound)
+    self.settings = {}
```

Comment:
this variable is set to None in the init method, do you have specific reason for that? Should the default to dict?

Reply:
it is (line 50)